### ISSUE TITLE
Auto ignoring some common-ish framework files

### ISSUE THREAD

**Issue by Jym77** [webpro/knip#148]
When running `knip` on a freshly created project (using yarn, which default to PnP), I get the following report:
```
Unused files (3)
.pnp.cjs
.pnp.loader.mjs
.yarn/releases/yarn-3.6.0.cjs
```

When running on a larger project, I also get reported as unused files:
```
.yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs
```
(and, I guess, more plugins would be reported)

---
I guess the way is just to ignore these in my config.
And I understand this may be a hotly debated topic as to which files are common enough that they should be ignored by default (or via a plugin?) Typically, I also get reported an unused `config/foo.ts` for another package that I use, but I consider this specific and fully agree it's on me to ignore it manually.

Maybe knip could have a yarn plugin, activated when `package.json#packageManager` points to yarn 🤔 Ideally, that plugin would read and understand `.yarnrc.yml`. Typically, I'd love to have reported an old version of yarn that I forgot to remove if `.yarnrc.yml#yarnPath` points to something else. 
`.yarnrc.yml` should contain links to yarn release file, plugins, and hint whether this installation is using PnP or `node_modules`.

**Comment by webpro** [comment]
Thanks for this idea and your suggestions, I like it. Will pick it up soon.

**Comment by webpro** [comment]
> I guess the way is just to ignore these in my config.

For now I'm going to recommend this indeed. The way Yarn PnP does all of this goes a bit against some conventions, e.g. with the complicated `.gitignore` settings and adding files inside a "dotdirectory" to VCS.

> Maybe knip could have a yarn plugin, activated when `package.json#packageManager` points to yarn 🤔 Ideally, that plugin would read and understand `.yarnrc.yml`. Typically, I'd love to have reported an old version of yarn that I forgot to remove if `.yarnrc.yml#yarnPath` points to something else.

There are other tools available to check whether versions are up-to-date, etc.

>  `.yarnrc.yml` should contain links to yarn release file, plugins, and hint whether this installation is using PnP or `node_modules`.

Currently this isn't something I'm using myself or see happening a lot. I could definitely be wrong, though. Happy to re-consider further down the road when PnP gains more traction (I also saw pnpm supports it).

Sorry, honestly I had different ideas when I first read your post. But since Knip plugins currently are not meant to cq don't support to ignore files, and ignoring custom files is normal, I'm going to leave it at that for now.

**Comment by webpro** [comment]
> Note that this is not just for yarn pnp. Yarn with `node_modules` also has its release and plugins reported by knip.

They can also be ignored by using e.g. `ignore: ".yarn/**"`, right?

But yeah I get your point. Knip is all about minimizing the amount of configuration necessary for any JS/TS project, so this is definitely something I'll keep in mind.

**Comment by Jym77** [comment]
> > Note that this is not just for yarn pnp. Yarn with `node_modules` also has its release and plugins reported by knip.
> 
> They can also be ignored by using e.g. `ignore: ".yarn/**"`, right?

Yes, absolutely.
They are also auto-ignored by setting the `project` config (of the top-level workspace) to `src` or wherever source files are kept.

OTOH, that means that older releases wouldn't be detected by knip after upgrading yarn version. But that is minor.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.19.0. See [Release 2.19.0](https://github.com/webpro/knip/releases/tag/2.19.0) for release notes._

**Comment by webpro** [comment]
Saw that we only need to ignore `.yarn`, which will then not reference the `.pnp.cjs` and `.pnp.loader.mjs` scripts, so I just went ahead with only that.
